desktop_version/src/Editor.cpp

```diff
@@ -1881,12 +1906,13 @@ void editorrenderfixed(void)
             ghost.realcol = graphics.getcol(ghost.col);
         }
 
-        if (ed.currentghosts + 1 < (int) ed.ghosts.size()) {
+        ed.currentghosts++;
+        if (ed.zmod)
+        {
             ed.currentghosts++;
-            if (ed.zmod) ed.currentghosts++;
-        } else {
-            ed.currentghosts = (int) ed.ghosts.size() - 1;
         }
+
+        ed.currentghosts = SDL_max(ed.currentghosts, ed.ghosts.size());
```

Comment:
Here's a useful mnemonic when working with `max` and `min` functions: The `max` function *max*imizes the numbers given (i.e. whichever is bigger), and the `min` function *min*imizes the numbers given (i.e. whichever is smaller).

In this case, `SDL_max` is the wrong function to use, because `ed.ghosts.size()` here will always be bigger no matter what. This results in ghosts always appearing all at once instead of appearing one at a time. Use `SDL_min` instead.
```suggestion
        ed.currentghosts = SDL_min(ed.currentghosts, ed.ghosts.size());
```